### DISCUSSION TITLE
Issues with android Touchable Opacities

### DISCUSSION THREAD

**Discussion by iMaupin** [wix/Detox#3904] [Q&A]
Recently, my test suite has broken on android devices. I have not made any changes to the test suite itself. I have noticed, however, that some of the TouchableOpacities, when tapped, appear to have been tapped (the UI responds, and the button changes color) but the effects of the button press are not triggered (e.g., the device does not navigate to a new screen). When tested manually the TouchableOpacities in question function as expected.

I did recently update one of my test devices to android 13, but confirmed the issue is also happening on android 12. I am using detox 19.13.0. 

Does anybody know what could be causing this change in behavior? These TOs have worked as expected in the past and I have not touched them in months.  The iOS versions of the tests have no issues with these components, and function as expected.

EDIT:  This appears to be the same issue reported here: [https://github.com/wix/Detox/issues/3879](url)
As OP states, substituting the .longPress() method appears to work, but retooling my entire suite to longPress() buttons on android is not a viable solution to this issue. Please advise.

**Comment by portyu9** [comment]
Hi @iMaupin 

This was a known Detox Android tap bug rather than a problem with your `TouchableOpacity` or `testID`.

The behavior you described — the control visually entering its pressed state but `onPress` not firing on Android while iOS works — was tracked in Detox issue #3879 and then closed as a duplicate of #3762.

The `.longPress()` workaround mentioned at the time was only temporary. The underlying #3762 issue was later fixed and closed, and users confirmed that the patched version resolved the problem.

You're using Detox `19.13.0`, so the solution today is to **upgrade Detox**, completely rebuild both the application APK and AndroidTest APK, and retest:

```text
app APK
AndroidTest APK
```

I would not refactor the application or replace normal `.tap()` calls with `.longPress()`.

If the same behavior still reproduces after upgrading to a current Detox version and rebuilding on a supported React Native/Android combination, that would be a new regression rather than the original known bug.

Please mark this answer as accepted if it helped, thank you.
